Source/JavaScriptCore/wasm/WasmOperations.cpp

```diff
@@ -609,10 +627,17 @@ JSC_DEFINE_JIT_OPERATION(operationIterateResults, void, (Instance* instance, con
             unboxedValue = bitwise_cast<uint64_t>(value.toNumber(globalObject));
             break;
         default: {
-            if (isFuncref(returnType) || isExternref(returnType)) {
-                if (isFuncref(returnType) && !value.isCallable()) {
-                    throwTypeError(globalObject, scope, "Funcref value is not a function"_s);
-                    return;
+            if (isExternref(returnType))
+                ASSERT(returnType.isNullable());
+            else if (isFuncref(returnType)) {
+                ASSERT(returnType.isNullable());
+                if (isFuncref(returnType)) {
```

Comment:
This is redundant: we already checked it at line 632.

Reply:
Fixed.